server/graphql/types/src/types/program/vaccination_card.rs

```diff
@@ -0,0 +1,102 @@
+use async_graphql::*;
+
+use chrono::NaiveDate;
+
+use dataloader::DataLoader;
+use graphql_core::{loader::StockLineByIdLoader, ContextExt};
+use service::vaccination::get_vaccination_card::{VaccinationCard, VaccinationCardItem};
+
+use crate::types::StockLineNode;
+
+pub struct VaccinationCardNode {
+    pub vaccination_card: VaccinationCard,
+}
+
+#[Object]
+impl VaccinationCardNode {
+    pub async fn id(&self) -> &str {
+        &self.vaccination_card.enrolment.program_row.id
+    }
+
+    pub async fn program_name(&self) -> &str {
+        &self.vaccination_card.enrolment.program_row.name
+    }
+
+    pub async fn patient_first_name(&self) -> &Option<String> {
+        &self.vaccination_card.enrolment.patient_row.first_name
+    }
+
+    pub async fn patient_last_name(&self) -> &Option<String> {
+        &self.vaccination_card.enrolment.patient_row.last_name
+    }
+
+    pub async fn items(&self) -> Vec<VaccinationCardItemNode> {
+        self.vaccination_card
+            .items
+            .iter()
+            .map(|item| VaccinationCardItemNode::from_domain(item.clone()))
+            .collect()
+    }
+}
+
+impl VaccinationCardNode {
+    pub fn from_domain(vaccination_card: VaccinationCard) -> VaccinationCardNode {
+        VaccinationCardNode { vaccination_card }
+    }
+}
+
+pub struct VaccinationCardItemNode {
+    pub item: VaccinationCardItem,
+}
+
+#[Object]
+impl VaccinationCardItemNode {
+    pub async fn id(&self) -> &str {
+        &self.item.row.id
+    }
+    pub async fn vaccine_course_dose_id(&self) -> &str {
+        &self.item.row.vaccine_course_dose_id
+    }
+    pub async fn vaccination_id(&self) -> &Option<String> {
+        &self.item.row.vaccination_id
+    }
+    pub async fn label(&self) -> &str {
+        &self.item.row.label
+    }
+    pub async fn min_age_months(&self) -> &f64 {
+        &self.item.row.min_age
+    }
+    pub async fn min_interval_days(&self) -> &i32 {
+        &self.item.row.min_interval_days
+    }
+    pub async fn given(&self) -> &Option<bool> {
+        &self.item.row.given
+    }
```

Comment:
Now I think about it, do we actually need the "given" field at all? It's implied by the `vaccination_date`, isn't it? I'd be inclined to remove it -- even if we want to return that field to front-end, we should just derive it rather than store it separately.

Reply:
No its slightly different: vaccination_date is set on all vaccination "events", where you can record whether it was administered or refused (given true/false). Given would be undefined/None before the event 👍

Reply:
Okay, and that's different to the encounter date that it's linked to?

Reply:
Yep 🫠 It probably will be mostly the same, but the clinician might enter historic vaccinations during an encounter

Reply:
Okay, but would they really ever enter historical non-vaccinations (i.e. a date but not "given")?

Reply:
🤷‍♀️  Maybe catching up on a backlog, if they we're entering at the time? Probably pretty unlikely though 👍 But regardless, the given true/false on a `vaccination` serves a purpose for non-historical vaccinations, and I don't think the given field should mean different things based on the date